src/lib/components/collections/req-res-section/sub-components/sub-components-header/CodeMirrorInput.svelte

```diff
@@ -0,0 +1,223 @@
+<script lang="ts">
+  import { afterUpdate, onMount } from "svelte";
+  import { EditorView } from "codemirror";
+  import { EditorState, Compartment, EditorSelection } from "@codemirror/state";
+  import { InputTheme, type AggregateEnvironment } from "./InputEditorTheme";
+  import { javascriptLanguage } from "@codemirror/lang-javascript";
+  import {
+    keymap,
+    ViewPlugin,
+    MatchDecorator,
+    Decoration,
+    placeholder,
+    hoverTooltip,
+    type Tooltip,
+  } from "@codemirror/view";
+  import { environmentHoverHighlightStyle } from "./EnvironmentHighlight";
+  export let currentTabId: string;
+  export let rawValue: string;
+  export let handleRawChange: () => void;
+  export let handleFocusChange: () => void;
+  export let handleBlurChange: () => void;
+  export let handleInputChange: (data: string) => void;
+  export let handleKeyUpChange: (e: EditorSelection) => void;
+  export let handleKeyDownChange: (e: KeyboardEvent) => void;
+  export let codeMirrorEditorDiv: HTMLDivElement;
+  export let filterData: AggregateEnvironment[];
+  export let handleEnvironmentBox: (change: boolean, envKey: string) => void;
+  let localEnvKey = "";
+
+  const ENVIRONMENT_REGEX = /({{[a-zA-Z0-9-_]+}})/g;
+
+  const ENV_HIGHLIGHT = "env-highlight";
+  const ENV_HIGHLIGHT_FOUND = "env-found";
+  const ENV_HIGHLIGHT_NOT_FOUND = "env-not-found";
+  let selectedTabId = currentTabId;
+  const languageConf = new Compartment();
+  let codeMirrorView: EditorView;
+  const updateExtensionView = EditorView.updateListener.of((update) => {
+    const userInput = update.state.doc.toString();
+    handleInputChange(userInput);
+    handleRawChange();
+    handleHighlightClass();
+  });
+  const keyBinding = keymap.of([
+    {
+      key: "Enter",
+      run: (view) => {
+        return true;
+      },
+    },
+  ]);
+
+  const handleEventsRegister = EditorView.domEventHandlers({
+    blur: (event, view: EditorView) => {
+      handleBlurChange();
+    },
+    focus: (event, view: EditorView) => {
+      handleFocusChange();
+    },
+    keyup: (event, view: EditorView) => {
+      handleKeyUpChange(codeMirrorView.state.selection);
+    },
+
+    keydown: (event, view: EditorView) => {
+      handleKeyDownChange(event);
+    },
+  });
+
+  const handleHighlightClass = () => {
+    const boxes = document.getElementsByClassName("env-not-found");
+    if (boxes.length > 0) {
+      for (const box of boxes) {
+        box.addEventListener("click", function handleClick(event) {
+          if (!filterData.find((k: { key: string }) => k.key === localEnvKey)) {
```

Comment:
let's move the filter statement inside a variable, outside the if condition and use that variable inside the if. Just to make it more readable.